src/virtualenv/discovery/cache.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from typing_extensions import Protocol
+else:
+    Protocol = object
```

Comment:
Note like this; this should be something like:

```
if sys.version_info >= (3, 10):
   from typing import Protocol
else:
   from typing_extensions import Protocol
```

with appropiate covdefaults pragma no covers.

Reply:
actually I'll remove all that since Protocol shipped in 3.8, ok?